props/prop_base.h

```diff
@@ -580,23 +583,31 @@ class PropBase : CommandParser, Looper, protected SaberBase, public ModeInterfac
 #ifdef BLADE_ID_SCAN_MILLIS
 #ifndef SHARED_POWER_PINS
 #warning SHARED_POWER_PINS is recommended when using BLADE_ID_SCAN_MILLIS
+#endif
+#ifndef BLADE_ID_SCAN_TIMEOUT
+#define BLADE_ID_SCAN_TIMEOUT 60 * 10 * 1000
 #endif
   bool find_blade_again_pending_ = false;
   uint32_t last_scan_id_ = 0;
-  bool ScanBladeIdNow() {
-    uint32_t now = millis();
-    if (now - last_scan_id_ > BLADE_ID_SCAN_MILLIS) {
-      last_scan_id_ = now;
-      size_t best_config = FindBestConfig();
-      if (current_config != blades + best_config) {
-  // We can't call FindBladeAgain right away because
-  // we're called from the blade. Wait until next loop() call.
-  find_blade_again_pending_ = true;
-      }
-      return true;
+    bool ScanBladeIdNow() {
+        uint32_t now = millis();
+        if (IsOn() == false && (now - blade_id_scan_start_) < BLADE_ID_SCAN_TIMEOUT) {
```

Comment:
Why is IsOn() == false here?
That will make blade scanning while on not work properly.

Reply:
The assumption was that a blade config file wouldn't change will the blade was ignited, as a result only search for a blade if the lightsaber is off and the timeout hasn't been reached. If you would like I can just remove the IsOn() check. The resulting if statement would be `if ((now - blade_id_scan_start_) < BLADE_ID_SCAN_TIMEOUT)`

Reply:
FWIW I think shutting off BLADE_ID_SCAN_MILLIS when the blade is lit is a good idea. In my limited experiments with it, I've found that scan values can change quite significantly when the blade is lit - particularly if using SnapshotID - resulting the array constantly trying to change when you don't want it to. Indeed I would go further and add a small delay between shutting off the blade and restarting BLADE_ID_SCAN_MILLIS to ensure that unusual scan values don't cause further unwanted array switches.